packages/backend-common/src/reading/GitlabUrlReader.ts

```diff
@@ -106,25 +106,28 @@ export class GitlabUrlReader implements UrlReader {
     // ref is an empty string if no branch is set in provided url to readTree.
     const branch = ref || projectGitlabResponseJson.default_branch;
 
-    // Fetch the latest commit in the provided or default branch to compare against
-    // the provided sha.
-    const branchGitlabResponse = await fetch(
+    // Fetch the latest commit that modifies the the filepath in the provided or default branch
+    // to compare against the provided sha.
+    const branchQuery = `ref_name=${branch}`;
+    const pathQuery = !!filepath ? `&path=${filepath}` : '';
```

Comment:
should construct these using `URLSearchParams` instead, which also solves the proper safe quoting of values

Reply:
Updated